internal/compiler/generator/cpp.rs

```diff
@@ -3622,6 +3622,10 @@ fn compile_builtin_function_call(
             ctx.generator_state.conditional_includes.cmath.set(true);
             format!("std::exp({})", a.next().unwrap())
         }
+        BuiltinFunction::Sign => {
+            ctx.generator_state.conditional_includes.cmath.set(true);
+            format!("((({0}) == 0.0) ? 0.0 : (std::signbit({0}) ? -1 : 1))", a.next().unwrap())
```

Comment:
There is also std::copysign(1, {}) but i don't know which one is better.

But the main problem here is that it evaluates its argument twice.
So  

This has the problem that it evaluates its argument twice.
So maybe i can suggest something like so:

```suggestion
            format!("[](auto a){ return a == 0 ? 0 : a < 0 ? -1 : 1; }({})", a.next().unwrap())
```